### PR TITLE
remove rebuilds and buildconfigs

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [n/a] JSON/YAML configuration changes are updated in the relevant schema
- [n/a] Changes to metadata also update the documentation for the metadata
- [n/a] Pull request has a link to an osbs-docs PR for user documentation updates

### CODE REVIEW

File: osbs/build/build_requestv2.py [hunk 1]
```diff
@@ -113,7 +112,7 @@ def render_custom_strategy(self):
         custom_strategy['from']['name'] = self.user_params.build_image
 
     def render_name(self):
-        """Sets the Build/BuildConfig object name"""
+        """Sets the Build object name"""
```
Comment (chmeliik) [inline osbs/build/build_requestv2.py:115]: Wait, so we're removing BuildConfigs but keeping Builds?

Comment (MartinBasti) [reply osbs/build/build_requestv2.py:115]: just for a while, to make rewrite to tekton pipelines easier

**Review by chmeliik — APPROVED** [review]
LGTM

Not gonna lie, I have no idea if this removes everything that should be removed, but the things that *did* get removed make sense to me... :smile: 

Just a nitpick about the commit message: imagestreams were removed too, not just rebuilds and buildconfigs. And next time, I would love if imagestreams, rebuilds and buildconfigs could be removed in 3 separate commits (if possible), but for this PR that's not necessary

**Comment by lkolacek** [comment]
The removed code looks reasonable. I also cannot tell if that's all. :) Considering the number of changes in this PR I would say it deserves more than 2 approvals.

**Review by lkolacek — APPROVED** [review]

File: osbs/api.py [hunk 2]
```diff
@@ -1202,117 +863,15 @@ def get_image_stream_tag(self, tag_id):
     def get_image_stream_tag_with_retry(self, tag_id):
         return self.os.get_image_stream_tag_with_retry(tag_id)
```
Comment (mkosiarc) [inline osbs/api.py:864]: what about this one

File: osbs/api.py [hunk 3]
```diff
@@ -1202,117 +863,15 @@ def get_image_stream_tag(self, tag_id):
...
-        with open(img_stream_tag_file) as f:
-            tag_template = json.load(f)
-
-        return self.os.ensure_image_stream_tag(stream, tag_name, tag_template,
-                                               docker_image_repo, scheduled,
-                                               insecure=insecure)
-
     @osbsapi
     def get_image_stream(self, stream_id):
         return self.os.get_image_stream(stream_id)
```
Comment (mkosiarc) [inline osbs/api.py:868]: or this ?

File: osbs/core.py [hunk 4]
```diff
@@ -52,7 +47,6 @@
 
 OCP_RESOURCE_API_VERSION_MAP = {
     'builds': OCP_BUILD_API_V1,
-    'buildconfigs': OCP_BUILD_API_V1,
     'imagestreams': OCP_IMAGE_API_V1,
```
Comment (mkosiarc) [inline osbs/core.py:50]: this?

File: osbs/core.py [hunk 5]
```diff
@@ -883,55 +734,6 @@ def get_image_stream_tag_with_retry(self, tag_id):
...
-            logger.debug('setting importPolicy.scheduled to: %s', scheduled)
-            changed = True
-
-        if changed:
-            logger.debug('modifying image stream tag: %s : %s', tag_id, tag)
-            self.put_image_stream_tag(tag_id, tag)
-
-        return changed
-
     def get_image_stream(self, stream_id):
```
Comment (mkosiarc) [inline osbs/core.py:737]: this?

File: osbs/api.py (hunk 2 above)
Comment (mkosiarc) [reply osbs/api.py:864]: it was explained to me by robert that we are keeping the get methods for now

**Review by mkosiarc — APPROVED** [review]

**Comment by mkosiarc** [comment]
code coverage dropped, but that is expected so I marked it was passing in the checklist

**Comment by rcerven** [comment]
rn updated

**Pull request merged**